dev/tests/api-functional/testsuite/Magento/GraphQl/GiftMessage/Cart/GiftMessageTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\GiftMessage\Cart;
+
+use Exception;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\GraphQl\Quote\GetMaskedQuoteIdByReservedOrderId;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+class GiftMessageTest extends GraphQlAbstract
+{
+    /**
+     * @var GetMaskedQuoteIdByReservedOrderId
+     */
+    private $getMaskedQuoteIdByReservedOrderId;
+
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->getMaskedQuoteIdByReservedOrderId = $objectManager->get(GetMaskedQuoteIdByReservedOrderId::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/GiftMessage/_files/quote_with_message.php
+     * @throws NoSuchEntityException
+     * @throws Exception
+     */
+    public function testGiftMessageForCart()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId->execute('message_order_21');
+        $query = <<<QUERY
+{
+    cart(cart_id: "$maskedQuoteId") {
```

Comment:
Please, create a separate private method i.e. `requestCartAndAssertResult` and move the duplicated code fragment there

Reply:
**Done**